kinto/core/storage/postgresql/__init__.py

```diff
@@ -737,14 +739,30 @@ def _format_conditions(self, filters, id_field, modified_field,
 
             if is_like_query:
                 # Operand should be a string.
-                # Add implicit start/end wildchars if none is specified.
+                # Add implicit start/end wildcards if none is specified.
                 if '*' not in value:
                     value = '*{}*'.format(value)
                 value = value.replace('*', '%')
 
             if filtr.operator == COMPARISON.HAS:
                 operator = 'IS NOT NULL' if filtr.value else 'IS NULL'
                 cond = '{} {}'.format(sql_field, operator)
+            elif filtr.operator in (COMPARISON.CONTAINS, COMPARISON.CONTAINS_ANY):
+                # Safely escape value. MISSINGs get handled below.
+                value_holder = '{}_value_{}'.format(prefix, i)
+                value = self.json.loads(value)
+                holders[value_holder] = value
+
+                array_type = "TEXT"
+                if len(value) > 0 and isinstance(value[0], int):
+                    array_type = "INT"
+
+                sql_operator = operators.setdefault(filtr.operator,
+                                                    filtr.operator.value)
+                # Thanks https://dba.stackexchange.com/a/157186
+                cond = "TRANSLATE(({})\:\:jsonb\:\:text, '[]','{}')\:\:{}[] {} :{}".format(
+                    sql_field, '{}', array_type, sql_operator, value_holder)
```

Comment:
I think this is all way too complicated and I am not crazy about the use of string processing to convert from one type to another. I think you should be converting the filter value to JSONB. This way we can support checking for fields containing complex objects like `[1, 2]` (in a field like `[[1,2], [5, 9]]`). There's no way to do "overlap" testing with this type, but I think that's OK -- do we really need `contains_any` at this point in time or did you just figure it was easy to implement at the same time?

Reply:
> do we really need `contains_any` 

Yes I do

Reply:
> I think this is all way too complicated and I am not crazy about the use of string processing to convert from one type to another.

That's the only way I found so far to be able to add it in the WHERE clause.

We could use `jsonb_to_text_array` but it would be have to be a JOIN.

If you find another way of doing it, I would be pleased to use it.

It was also a bit complex to use the SQLAlchemy execute parameters with an array.